hera_cal/vis_clean.py

```diff
@@ -7,29 +7,240 @@
+        for rownum, wrow in enumerate(wout):
+            if ax == 'both':
+                if np.any(np.isclose(wout[rownum, :min_flag_edge_distance[1]], 0.0)) | np.any(np.isclose(wout[rownum, -min_flag_edge_distance[1] - 1:], 0.0)):
+                    wout[rownum, :] = 0.
+            elif np.any(np.isclose(wout[rownum, :min_flag_edge_distance], 0.0)) | np.any(np.isclose(wout[rownum, -min_flag_edge_distance - 1:], 0.0)):
+                wout[rownum, :] = 0.
+        if ax == 'both':
+            wout = flag_rows_with_flags_within_edge_distance(wout, min_flag_edge_distance[0], ax='time')
+    return wout
+
+
+def flag_rows_with_contiguous_flags(weights_in, max_contiguous_flag, ax='freq'):
+    """
+    flag any row or column with contiguous zero-weights over limit.
+
+    Parameters
+    ----------
+    weights_in : array-like, 2d
+        weights to check. any row (ax='time') or col (ax='freq')
+        with contiguous regions of zero with length greater then max_contiguous_flag
+        will be set to zero.
+    max_contiguous_flag : integer (or 2-list/tuple)
+        any row or column with contiguous weights equal to zero greater or equal to this value
+        will be set entirely to zero.
+    ax : str, optional
+        axis to perform flagging over.
+        valid options include 'time' (zeroth axis), 'freq'(1 axis)
+        or 'both'.
+        default is 'freq'
+    """
+    if ax == 'time':
+        wout = flag_rows_with_contiguous_flags(weights_in.T, max_contiguous_flag).T
+    else:
+        wout = copy.deepcopy(weights_in)
+        for rownum, wrow in enumerate(wout):
+            max_contiguous = 0  # keeps track of the largest contig flags in integration.
+            current_flag_length = 0  # keeps track of current contig flag size.
+            on_flag = False  # keep track if currently on a flag.
+            # iterate over each channel in integration.
+            for wr in wrow:
+                on_flag = (wr == 0)  # if weights are zero, on a flag.
+                # if on a flag, +1 current flag size.
+                if on_flag:
+                    current_flag_length += 1
+                else:
+                    # otherwise, check if the current flag len > max.
+                    # update max_contiguous if appropriate.
+                    if current_flag_length >= max_contiguous:
+                        max_contiguous = current_flag_length
+                    current_flag_length = 0
+            if ax == 'both':
+                if max_contiguous >= max_contiguous_flag[1]:
+                    wout[rownum][:] = 0.
+            elif max_contiguous >= max_contiguous_flag:
+                wout[rownum][:] = 0.
+        if ax == 'both':
+            wout = flag_rows_with_contiguous_flags(wout, max_contiguous_flag[0], ax='time')
+    return wout
+
+
+def get_max_contiguous_flag_from_filter_periods(x, filter_centers, filter_half_widths):
+    """
+    determine maximum contiguous flags from filter periods
```

Comment:
perhaps we can be a little more verbose here, just b/c its such a niche function.
1. you mention filter periods in the doc but filter periods is not an input.
2. why is a maximum contiguous flag needed? This seems specific to a DPSS / linear filter approach, perhaps you can clarify briefly, thanks

Reply:
I've found that when performing linear inpainting with both DPSS and DFT modes, you tend to get  artifacts inside of gaps that are significantly larger then the smallest wavelength mode that is being used to interpolate. 

A more classical example that you might be familiar with is when you try to fit a high order polynomial to data with large gaps, you can end up getting nonsense structures inside of the gaps.

Reply:
I think this arises because there are degeneracies of what should be inside of the large gap given the SNR of the data being fitted. This makes me think that there is probably a better way to fit the modes that we should be taking advantage of but better priors for how the gap structure should behave (beyond that it is composed of the linear sum of our fitting basis) for now, subject of future work ;)

Reply:
Okay, so the result you propose is to flag the entire channel or integration? Or to just leave that stretch of channels (or integrations) flagged?

Reply:
That's correct. If we are doing frequency axis filtering and there is a contiguous chunk of channels that are flagged that exceeds the wavelength of the smallest sinusoid wavelength being fitted then we flag that integration. Similarly when filtering with time.